.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,13 @@
+
+
+
+
+
+
+
+---
+**^Add meaningful description above**
+
+By opening a pull request here, you confirm that the changes are all your own work, and provided to Apache NetBeans under the terms and conditions of the Apache License, without any additional terms or conditions (see also section 5 of the license).
```

Comment:
This should be Apache License Version 2.0 and IMHO the more important part is, something along the lines:

> You acknowledge, that you donates the change to the ASF, which means, that the ASF can relicense and sublicense the work.

This would become important, if at anytime the ASF decides to release the Apache License 2.1 or 3.0. I don't know the ration, but I assume, that 99% of the repository is ASF-2.0-ASL (with donation clause).

Reply:
Searching for the right legal wording, the ICLA itself looks fitting:

> 1. You grant to the Apache Software Foundation (Foundation) and to recipients of software distributed by the Foundation a perpetual, worldwide, non-exclusive, no-charge, royalty-free, irrevocable copyright license to reproduce, prepare derivative works of, publicly display, publicly perform, sublicense, and distribute Your Contributions and such derivative works.
> 2. You grant to the Foundation and to recipients of software distributed by the Foundation a perpetual, worldwide, non-exclusive, no-charge, royalty-free, irrevocable (except as stated in this section) patent license to make, have made, use, offer to sell, sell, import, and otherwise transfer the Work, where such license applies only to those patent claims licensable by You that are necessarily infringed by Your Contribution(s) alone or by combination of Your Contribution(s) with the Work to which such Contribution(s) was submitted. If any entity institutes patent litigation against You or any other entity (including a cross-claim or counterclaim in a lawsuit) alleging that your Contribution, or the Work to which you have contributed, constitutes direct or contributory patent infringement, then any patent licenses granted to that entity under this Agreement for that Contribution or Work shall terminate as of the date such litigation is filed.
> 3. You represent that you are legally entitled to grant the above license. If your employer(s) has rights to intellectual property that you create that includes your Contributions, you represent that you have received permission to make Contributions on behalf of that employer, that your employer has waived such rights for your Contributions to the Foundation, or that your employer has executed a separate Corporate CLA with the Foundation.
> 4. You represent that each of Your Contributions is Your original creation. You represent that Your Contribution submissions include complete details of any third-party license or other restriction (including, but not limited to, related patents and trademarks) of which you are personally aware and which are associated with any part of Your Contributions.

Reply:
@matthiasblaesing ASF don't require anything from a legal perspective! Just opening a PR is enough. That's why section 5 is in the license. You cannot work with the sources at all without accepting the licence, and therefore have already accepted the terms of contribution.

This is about better information if we have it at all. The wording should be from or refer to that license section, nothing more, or we actually muddy the waters.

Agree should be Apache License 2.0.

Reply:
I'm not a lawyer and ask myself why on earth I had to get an ICLA on file, but ok, I made my point, so fine with me.

Reply:
@matthiasblaesing committers need an ICLA, contributors don't. In general at least - we may ask for one if it's a large contribution, unsure of provenance, etc.

Although this is not official, it links to some useful stuff on ASF side and has been discussed here https://apetro.ghost.io/apache-contributors-no-cla/

Reply:
@neilcsmith-net, one thing we may want is a proper  name/email address. Otherwise we lose track of authorship, right?

Reply:
Thanks @vieiro Yes, a point about that might be worth including as well as it comes up a lot.  I'm wary of putting too much text here though.  Will look at revising with this and Matthias' comments. Some things really should go in a revised .github/CONTRIBUTING.md file, and probably move that to top level.

Aside - the commits@ list is meant to be the canonical record of contributions as far as I know.  Interesting that the GitHub / ASF integration seems to partly redact the author's email there.

Reply:
I've pushed an update.  Have removed reference to NetBeans or ASF explicitly though to try and keep the length down.  This is meant to be info rather than legal (which is covered by section 5 ASL or CLA), so think it's obvious enough.

Also added line about author info.

I'm not convinced we have to have this at all, but it's an option we have if we want to have that info more obvious.  Shame no form templates like issues yet though!

Reply:
Cool!

Since this is an .md (markdown) file, maybe we want to add a link to the contribution guidelines https://netbeans.apache.org/participate/index.html ? Maybe a link to the dev mailing https://netbeans.apache.org/community/mailing-lists.html ?

The idea is to let people know how to do things without having to Google how first :-)

Reply:
D'oh. I'm confused myself. We have at least two different contribution guidelines. Here's another one https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=74681408

Reply:
@vieiro while it's a markdown file (as copied and adapted from Airflow) it's not like the new issue templates in #3415 where we can have actual markup displayed.  See the image, it's plain text.  I tried links like Airflow has in theirs, but decided it looks far better and easier to read without.

There is a direct link to the contributing guidelines file already under the text input, and various other places in the UI. This links to `.github/CONTRIBUTING.md`, which we could improve in place as well as link off (and as mentioned above move up next to README so it's not hidden in a checkout).

Reply:
Ok with me then. Full speed ahead!